src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1421,92 +1424,152 @@ bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,
             {
                 const wchar_t wch = *curr++;
 
                 if (_isHexNumber(wch))
                 {
-                    value *= 16;
+                    parameterValue *= 16;
                     unsigned int intVal = 0;
                     if (s_HexToUint(wch, intVal))
                     {
-                        value += intVal;
+                        parameterValue += intVal;
                     }
                     else
                     {
                         // Encountered something weird oh no
                         foundColor = false;
                         break;
                     }
-                    // If we're on the blue component, we're not going to see a /.
-                    // Break out once we hit the end.
-                    if (component == 2 && curr >= string.cend())
-                    {
-                        foundValidColorSpec = true;
-                        break;
-                    }
                 }
-                else if (wch == L'/')
+            }
+
+            if (isSharpSignFormat)
+            {
+                // Successfully parsed this component. Start the next one.
+                foundColor = true;
+            }
+            else
+            {
+                if (curr < string.cend() && *curr == L'/')
                 {
-                    // Break this component, and start the next one.
+                    // Same as above but need to skip the delimiter.
+                    curr++;
                     foundColor = true;
-                    break;
                 }
                 else
                 {
-                    // Encountered something weird oh no
+                    // Invalid delimiter. Treat it as an error.
                     foundColor = false;
                     break;
                 }
             }
+
             if (!foundColor || curr >= string.cend())
             {
-                // Indicates there was a some error parsing color
-                //  or we're at the end of the string.
+                // Indicates there was some error parsing color.
                 break;
             }
         }
+
+        if (curr >= string.cend())
+        {
+            // We're at the end of the string and we have successfully parsed the color.
+            foundValidColorSpec = true;
+        }
     }
+
     // Only if we find a valid colorspec can we pass it out successfully.
     if (foundValidColorSpec)
     {
-        DWORD color = RGB(LOBYTE(colorValues.at(0)),
-                          LOBYTE(colorValues.at(1)),
-                          LOBYTE(colorValues.at(2)));
+        for (size_t component = 0; component < 3; component++)
+        {
+            auto& parameterValue = til::at(parameterValues, component);
+            auto& colorValue = til::at(colorValues, component);
+
+            // Calculate the actual color value based on the hex digit count.

```

Comment:
Is there some magical trick that can simplify this?

Reply:
til:color probably could be used here

Reply:
You could do something like this:
```cpp
const auto scaleMultiplier = isSharpSignFormat ? 0x10 : 0x11;
const auto scaleDivisor = scaleMultiplier << 8 >> 4 * (4 - rgbHexDigitCount);
colorValue = parameterValue * scaleMultiplier / scaleDivisor;
```
And change the `colorValues` array back to `unsigned int`. That would also let you get rid of the casts below.